### ISSUE TITLE
Unable to load schema from a URL

### ISSUE THREAD

**Issue by tubbo** [dotansimha/graphql-eslint#525]
**Describe the bug**

When trying to load my schema from an external URL, I get an error coming from `node-fetch`:

```
TypeError: Expected signal to be an instanceof AbortSignal
    at new Request (/Users/tscott/Code/nerdst/node_modules/node-fetch/lib/index.js:1222:10)
    at Object.deserializeRequest (/Users/tscott/Code/nerdst/node_modules/sync-fetch/shared.js:45:10)
    at Socket.<anonymous> (/Users/tscott/Code/nerdst/node_modules/sync-fetch/worker.js:16:26)
    at Socket.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1241:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

This happens on every URL that I throw at it

**To Reproduce**

Load your schema from a URL in GraphQL config:

```yaml
schema: http://localhost:8000/graphql
documents: src/**/*.gql
```

**Expected behavior**

The schema from the URL I provided should be used

**Environment:**

- OS: macOS v11.3, though I've also seen this happen in CI
- `@graphql-eslint/...`: `"^1.1.2"`
- NodeJS: v12.22.1

**Additional context**

I did some digging and noticed that `node-fetch` is somehow wrong about its assertion that the passed-in signal is not an AbortSignal. Something strange is going on with that logic and I'm pretty sure it's the root cause of all this, but plenty of other packages I depend on use the exact same `node-fetch` version without any problems. GraphQL Code Generator, using the exact same graphql-config (`.graphqlrc.yml`), can introspect the graph at this URL just fine, but as soon as I try to run ESLint, that's when things start to break down and I get that error all over the place.

Also wanted to say how helpful this project has been. `eslint-plugin-graphql` was what we were using before and honestly I just disabled it after a while because it didn't provide any useful feedback. This plugin, on the other hand, not only fully validates our operations but also our schema, which has been very helpful in identifying unused code, formatting mistakes, and lack of documentation. Thanks for making it!

**Comment by B2o5T** [comment]
Hi @tubbo thank you very much! Your detailed report is very helpful! Your feedback is awesome ☺️.

I found where the issue comes, firstly from [@graphql-tools/url-loader](https://github.com/ardatan/graphql-tools/tree/master/packages/loaders/url) that uses [sync-fetch](https://www.npmjs.com/package/sync-fetch) under the hood.

It's seems like `sync-fetch` don't support aborting signals from fetch API, so when I comment [this line](https://github.com/ardatan/graphql-tools/blob/93f795f20ab52345e70180e6db9985f14deaf697/packages/loaders/url/src/index.ts#L352) I got from [this line](https://github.com/larsgw/sync-fetch/blob/b751461b9bd4229a29eb6db72a20bc5fe8404b00/index.js#L32)
```sh
SyntaxError: Unexpected token d in JSON at position 0
    at JSON.parse (<anonymous>)
    at fetch (/Users/dimitri/Desktop/pickpacknship/node_modules/sync-fetch/index.js:35:25)
    at executor (/Users/dimitri/Desktop/pickpacknship/node_modules/@graphql-eslint/eslint-plugin/node_modules/@graphql-tools/url-loader/index.cjs.js:177:39)
```
because response string contains some `deserializeRequest` data that throws while it's parse with `JSON.parse` in my side.

![image](https://user-images.githubusercontent.com/7361780/125065392-09e90a00-e0b2-11eb-815d-5b77f0109217.png)

**Comment by tubbo** [comment]
Thanks @B2o5T, that fixed it for me as well. Since `sync-fetch` doesn't seem to support this feature, I guess I should make a PR to `@graphql-tools/url-loader` so that this gets removed. Looking into the history it doesn't seem like this was directly intentional?

**Comment by tubbo** [comment]
hmm I'm a little confused here. the only time i get this error is when i'm using `@graphql-tools/url-loader` within `graphql-eslint`...

I made an example repo here https://github.com/tubbo/graphql-tools-example

If you run `yarn start` you don't get an error, but if you run `yarn lint` then you do.

**Comment by tubbo** [comment]
Looks like this is indeed a problem with the URL Loader so I'm closing this issue and progress will continue over on the other repo. Thanks for helping to dig in!

**Comment by talentlessguy** [comment]
I don't know why but it still happens in 1.1.4 for me :/

```
/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/node-fetch@2.6.1/node_modules/node-fetch/lib/index.js:1222
                        throw new TypeError('Expected signal to be an instanceof AbortSignal');
                        ^

TypeError: Expected signal to be an instanceof AbortSignal
    at new Request (/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/node-fetch@2.6.1/node_modules/node-fetch/lib/index.js:1222:10)
    at Object.deserializeRequest (/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/sync-fetch@0.3.0/node_modules/sync-fetch/shared.js:45:10)
    at Socket.<anonymous> (/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/sync-fetch@0.3.0/node_modules/sync-fetch/worker.js:16:26)
    at Socket.emit (node:events:406:35)
    at endReadableNT (node:internal/streams/readable:1331:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

/home/v1rtl/Coding/brandname/ur-db/pages/dash/posts.tsx
  0:0  error  Parsing error: Command failed: /home/v1rtl/n/bin/node /home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/sync-fetch@0.3.0/node_modules/sync-fetch/worker.js
/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/node-fetch@2.6.1/node_modules/node-fetch/lib/index.js:1222
                        throw new TypeError('Expected signal to be an instanceof AbortSignal');
                        ^

TypeError: Expected signal to be an instanceof AbortSignal
    at new Request (/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/node-fetch@2.6.1/node_modules/node-fetch/lib/index.js:1222:10)
    at Object.deserializeRequest (/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/sync-fetch@0.3.0/node_modules/sync-fetch/shared.js:45:10)
    at Socket.<anonymous> (/home/v1rtl/Coding/brandname/ur-db/node_modules/.pnpm/sync-fetch@0.3.0/node_modules/sync-fetch/worker.js:16:26)
    at Socket.emit (node:events:406:35)
    at endReadableNT (node:internal/streams/readable:1331:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

✖ 1 problem (1 error, 0 warnings)
```

Here's my `.graphqlrc.yml`:

```js
schema:
  - https://example.com/v1/graphql:
```

and `.eslintrc`:

```js
{
  "parser": "@typescript-eslint/parser",
  "extends": [
    "plugin:@typescript-eslint/recommended",
    "eslint:recommended",
  ],
  "plugins": ["prettier"],
  "parserOptions": {
    "ecmaVersion": 2020,
    "sourceType": "module"
  },
  "overrides": [
    {
      "files": ["*.tsx", "*.ts", "*.jsx", "*.js"],
      "processor": "@graphql-eslint/graphql"
    },
    {
      "files": ["*.graphql"],
      "parser": "@graphql-eslint/eslint-plugin",
      "plugins": ["@graphql-eslint"],
      "parserOptions": {
        "schema": "https://example.com/v1/graphql"
      }
    }
  ],
  "env": {
    "node": true,
    "es6": true
  },
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "off",
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/ban-ts-comment": "off",
    "no-unused-vars": "off",
    "no-console": "off",
    "no-undef": "off"
  }
}
```

and my package.json:

```json
{
  "name": "dashboard",
  "dependencies": {
    "@apollo/client": "^3.3.21",
    "graphql": "^15.5.1"
  },
  "devDependencies": {
    "@babel/core": "^7.14.6",
    "@graphql-eslint/eslint-plugin": "^1.1.4",
    "@types/node": "^16.3.3",
    "@typescript-eslint/eslint-plugin": "^4.28.3",
    "@typescript-eslint/parser": "^4.28.3",
    "apollo": "^2.33.4",
    "eslint": "^7.31.0",
    "ts-node": "^10.1.0",
    "typescript": "^4.3.5"
  },
  "scripts": {
    "dev": "next",
    "build": "next build",
    "start": "NODE_ENV=production next start",
    "generate:apollo": "apollo client:codegen --target typescript --outputFlat ./generated"
  }
}
```

**Comment by B2o5T** [comment]
Hi @talentlessguy, thank you for your report 😜
I think this is happening because `url-loader` was fixed, but not released yet, please ping @ardatan in this PR https://github.com/ardatan/graphql-tools/pull/3175

**Comment by B2o5T** [comment]
@tubbo @talentlessguy fixed in 2.0.0
